Ssl support

bhuvnesh17 opened the pull request:
Added support for SSL Connection

bhuvnesh17 committed:
version bump

bhuvnesh17 committed:
adding ssl support

bhuvnesh17 committed:
udpates to ssl

bhuvnesh17 committed:
ssl updates

bhuvnesh17 committed:
testing

bhuvnesh17 committed:
updated port on testing file

bhuvnesh17 committed:
Updated documentation

bhuvnesh17 committed:
changing port to an unused port for testing

ghost commented:
@bhuvnesh17  please remove references to Kafka extension from the documentation.

ghost commented:
Here's the code review 1 comments:

**Config.yml**
 - `metricPrefix`: the description of `metricPrefix` with and without sim. Please refer aws redshift extension for an example
 - if you are using appd-commons 2.2.0, please add sample dashboards and `healthChecksEnabled` flag


**Documentation**
Please add documentation for:
- env variables support in readme
- key generation and monitoring over SSL (examples in Kafka)
- how to use env variables in config.yml
- troubleshooting with jconsole (both with and without SSL)


 Next Steps:
- BTD

bhuvnesh17 committed:
documentation

bhuvnesh17 committed:
docs update

bhuvnesh17 committed:
docs

bhuvnesh17 committed:
updates to doc

bhuvnesh17 commented:
Updated metric prefix, added SSL generation examples. 
troubleshooting section added. 
No dashboard added as this is a generic extension. 
No env variables added as they are needed case by case

bhuvnesh17 committed:
adding mutual auth 1

bhuvnesh17 committed:
mutual auth 2

bhuvnesh17 committed:
working mutual auth

bhuvnesh17 committed:
updated config

bhuvnesh17 committed:
removing ssl stuff completely
